### PR TITLE
[2631] Add sites to course creation

### PR DESCRIPTION
**Pull request by faissalMT**
### Context
In order for https://github.com/DFE-Digital/manage-courses-frontend/pull/798 to function correctly the course needs to have site information.  

### Changes proposed in this pull request
Provide the provider sites when requesting a course object from `build_new()`.

### Guidance to review
We'll want to have an include parameter at some point, but I think this works for now.  
  
This also includes a fix for a test which was breaking if `config/azure_environments.yml` was present on your real filesystem.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by faissalMT** [comment]
[Implement back button for further ed providers](https://trello.com/c/39EnNzVs/2631-implement-back-button-for-further-ed-providers)

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@
 SimpleCov.start "rails" do
   add_filter "/mcb/"
 end
+SimpleCov.start do
```
Comment (defong) [inline spec/spec_helper.rb]: Why is this here?

File: spec/requests/api/v2/build_new_course_spec.rb [hunk 2]
```diff
@@ -31,8 +31,9 @@
         Subject: API::V2::SerializableSubject,
         PrimarySubject: API::V2::SerializableSubject,
         Provider: API::V2::SerializableProvider,
+        Site: API::V2::SerializableSite,
       },
-      include: %i[subjects sites provider accrediting_provider],
+      include: [:subjects, :sites, :accrediting_provider, :provider, provider: [:sites]],
```
Comment (defong) [inline spec/requests/api/v2/build_new_course_spec.rb]: Is this actually getting tested?

File: spec/spec_helper.rb (hunk 1 above)
Comment (faissalMT) [reply spec/spec_helper.rb]: Specs were being included as part of the coverage

File: spec/requests/api/v2/build_new_course_spec.rb (hunk 2 above)
Comment (faissalMT) [reply spec/requests/api/v2/build_new_course_spec.rb]: This is the test, if you remove this change the tests will fail.

Comment (defong) [reply spec/requests/api/v2/build_new_course_spec.rb]: i was kinda expecting a test that was like `returns a course with subject relationships`
